Gemfile.tt

```diff
@@ -17,6 +17,7 @@ gem "lograge"
 gem "okcomputer"
 gem "sentry-ruby"
 gem "sentry-rails"
+gem "net-http"
```

Comment:
Curious to why we have to specify this? (not saying it's wrong, just interested🙂)

Reply:
yeah I commented this why in the other PR but yeah this is to make the warnings disappear. David Rodriguez explained it here 
https://github.com/ruby/net-imap/issues/16 
I can remove it if that's not good enough reason :)

Reply:
Ah right I didn't notice that - I'm happy with keeping it in, though I think it could be worth having a comment linking to that issue to provide some context.

Reply:
History: Ruby has been slowly moving things out of it's stdlib and into gems which are installed by default. The idea being to decouple improving the libs from ruby releases.

net-http was moved out in 3.0

I have some questions about whether this would work with Ruby 2.7.x and older. I'm not saying we definitely need to care about that but we should document it if this template won't run on a supported Ruby. 

I'm also guessing that the thing that is actually using net-http will add it as a dependency soon so this line is probably not required long term. Is that correct? If so we should document that because otherwise people will be to scared to ever remove it from their Gemfiles :-)

Reply:
@emre-simtay I commented out this line and tried running the template under various versions of Ruby (see below). I only get the warnings on 2.7 and 3.1 is out now so I don't think we need to worry about warnings on 2.7 because it's unlikely we would start a new app on that version in 2022.

### ruby 2.7.3

```
✦ ❯ be rails s -p 3001
/Users/eoinkelly/.rbenv/versions/2.7.3/lib/ruby/2.7.0/net/protocol.rb:66: warning: already initialized constant Net::ProtocRetryError
/Users/eoinkelly/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/net-protocol-0.1.2/lib/net/protocol.rb:68: warning: previous definition of ProtocRetryError was here
/Users/eoinkelly/.rbenv/versions/2.7.3/lib/ruby/2.7.0/net/protocol.rb:206: warning: already initialized constant Net::BufferedIO::BUFSIZE
/Users/eoinkelly/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/net-protocol-0.1.2/lib/net/protocol.rb:208: warning: previous definition of BUFSIZE was here
/Users/eoinkelly/.rbenv/versions/2.7.3/lib/ruby/2.7.0/net/protocol.rb:503: warning: already initialized constant Net::NetPrivate::Socket
/Users/eoinkelly/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/net-protocol-0.1.2/lib/net/protocol.rb:504: warning: previous definition of Socket was here
=> Booting Puma
=> Rails 7.0.1 application starting in development
=> Run `bin/rails server --help` for more startup options
Puma starting in single mode...
* Version 4.3.10 (ruby 2.7.3-p183), codename: Mysterious Traveller
* Min threads: 5, max threads: 5
* Environment: development
* Listening on tcp://127.0.0.1:3001
* Listening on tcp://[::1]:3001
Use Ctrl-C to stop
```

### ruby 3.0.3

```
✦ ❯ be rails s -p 3001
=> Booting Puma
=> Rails 7.0.1 application starting in development
=> Run `bin/rails server --help` for more startup options
Puma starting in single mode...
* Version 4.3.10 (ruby 3.0.3-p157), codename: Mysterious Traveller
* Min threads: 5, max threads: 5
* Environment: development
* Listening on tcp://127.0.0.1:3001
* Listening on tcp://[::1]:3001
Use Ctrl-C to stop
```

### ruby 3.1.0

```
✦ ❯ be rails s -p 3001
Calling `DidYouMean::SPELL_CHECKERS.merge!(error_name => spell_checker)' has been deprecated. Please call `DidYouMean.correct_error(error_name, spell_checker)' instead.
Calling `DidYouMean::SPELL_CHECKERS.merge!(error_name => spell_checker)' has been deprecated. Please call `DidYouMean.correct_error(error_name, spell_checker)' instead.
=> Booting Puma
=> Rails 7.0.1 application starting in development
=> Run `bin/rails server --help` for more startup options
Puma starting in single mode...
* Version 4.3.10 (ruby 3.1.0-p0), codename: Mysterious Traveller
* Min threads: 5, max threads: 5
* Environment: development
* Listening on tcp://127.0.0.1:3001
* Listening on tcp://[::1]:3001
Use Ctrl-C to stop
```

Reply:
> I'm also guessing that the thing that is actually using net-http will add it as a dependency soon so this line is probably not required long term. Is that correct? 

In theory yes but [apparently there's something up with how bundler loads stdlib gems that means doing so would require dropping support for Ruby 2.5](https://github.com/lostisland/faraday-net_http/pull/5#issuecomment-854762189) so it's possible we won't see this happen for a while.

So +1 to having a comment explaining why it's there and to get us to check back on it in a year or so (if we decide to keep it as a dependency at all)

Reply:
> Ruby 2.7.0+ required, Ruby 3.0+ preferred
> https://edgeguides.rubyonrails.org/7_0_release_notes.html

Rails 7 won't run on very old Ruby so we don't need to worry about it.

Given that Ruby 3.1 is now available, I don't think we want to encourage people to create **new** Rails apps on Ruby 2.7 so I don't think we need this change at all. I think our preferred solution for folks hitting warnings on Ruby 2.7 would be to upgrade their Ruby version (this repos is for _new_ apps only so that should be an easy thing to do)

Reply:
Ok I am removing this to prevent us from getting confused in the future.